src/services/geometry/dd4hep/JDD4hep_service.cc

```diff
@@ -105,6 +107,19 @@ void JDD4hep_service::Initialize() {
         dd4hep::setPrintLevel(static_cast<dd4hep::PrintLevel>(print_level));
         LOG << "Loading DD4hep geometry from " << m_xml_files.size() << " files" << LOG_END;
         for (auto &filename : m_xml_files) {
+
+            // Check that this XML file actually exists.
+            if( ! std::filesystem::exists(filename) ){
+                // Here we go against the standard practice of throwing an error and print
+                // the message and exit immediately. This is because we want the last message
+                // on the screen to be that this file doesn't exist.
+                auto mess = fmt::format(fmt::emphasis::bold | fg(fmt::color::red),"ERROR: ");
+                mess += fmt::format(fmt::emphasis::bold, "file: {} does not exist!",  filename);
+                mess += "\nCheck that your DETECTOR and DETECTOR_CONFIG environment variables are set correctly.";
+                std::cerr << std::endl << std::endl << mess << std::endl << std::endl;
+                _exit(-1);
```

Comment:
Could you actually make it compatible with this functionality

https://github.com/eic/EICrecon/blob/ff19490a6bdd4c95a7d0cb9e6897e532e69905a8/src/services/geometry/dd4hep/JDD4hep_service.cc#L127

Described here:

https://eic.github.io/EICrecon/#/design/common_flags_env?id=flags

Reply:
Because it is very convenient to give just a file name, assuming DETECTOR_PATH is given. 

Like 

```
eicrecon ... -Pdd4hep:xml_files=epic.xml

# now try tracking only
eicrecon ... -Pdd4hep:xml_files=epic_tracking_only.xml

# or switching between major geometry variants
```

Reply:
If you are confused now, I can fix that (file I mean)

Reply:
I guess I am confused. The values in `m_xml_files` should already have the DETECTOR_PATH prepended (see line 79). Thus, it looks like the second try in the catch clause is just prepending it a second time. Or am I missing something?